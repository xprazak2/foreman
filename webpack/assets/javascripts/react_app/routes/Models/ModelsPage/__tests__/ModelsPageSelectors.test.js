import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  modelsAreLoading,
  modelsMessage,
  modelsHaveError,
  modelsHaveData,
} from '../ModelsPageSelectors';

const stateFactory = (obj) => ({ modelsPage: obj })

const fixtures = {
  'should return loading state': () =>
    modelsAreLoading(stateFactory({ isLoading: true })),
  'should return message': () =>
    modelsMessage(stateFactory({ message: { type: 'text', text: 'msg' }})),
  'should return whether page has error': () =>
    modelsHaveError(stateFactory({ hasError: true })),
  'should return whether page has data': () =>
    modelsHaveData(stateFactory({ hasData: true }))
}

describe('ModelsPageSelectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
