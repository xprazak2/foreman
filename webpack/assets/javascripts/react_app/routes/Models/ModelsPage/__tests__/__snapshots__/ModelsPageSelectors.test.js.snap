// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelsPageSelectors should return loading state 1`] = `true`;

exports[`ModelsPageSelectors should return message 1`] = `
Object {
  "text": "msg",
  "type": "text",
}
`;

exports[`ModelsPageSelectors should return whether page has data 1`] = `true`;

exports[`ModelsPageSelectors should return whether page has error 1`] = `true`;
